### PR TITLE
Adds boto3 models for AWS Lambda

### PR DESCRIPTION
**Pull request by abishekvashok**
Adds RCE sink annotations of AWS Lambda which is a boto3 client.

Requires explicit type annotations and mypy_boto3 package being used to
annotate the code as boto3.client() returns `typing.any` because of the
large number of clients it contain. boto3 doesn't have any classes to
model as well (every client is literally the same class!)

Test plan:
- run pyre analyze --no-verify against the following code:
```python3
import flask
import boto3
from flask import Flask, render_template
from mypy_boto3_lambda import LambdaClient

app = Flask(__name__)


@app.route("/rce/<string:payload>")
def definite_rce(payload: str) -> None:
    client: LambdaClient = boto3.client('lambda')
    response = client.create_function(
        FunctionName='string',
        Runtime='nodejs',
        Role='string',
        Handler='string',
        Code={
            'ImageUri': payload
        },
        Description='string',
        Timeout=123,
        MemorySize=123,
        Publish=True,
        ImageConfig={
            'EntryPoint': [
                payload,
            ],
            'Command': [
                payload,
            ],
            'WorkingDirectory': payload
        },
    )
    response = client.update_function_code(
        FunctionName='string',
        ZipFile=b'bytes',
        S3Bucket=payload,
        S3Key='string',
        S3ObjectVersion='string',
        ImageUri='string',
        Publish=True,
        RevisionId='string',
        Architectures=[
        'x86_64'|'arm64',
        ]
    )
```

with the following `.pyre_configuration`:
```json
{
  "source_directories": [
    "."
  ],
  "taint_models_path": "../../stubs",
  "search_path": [
    ".",
    {"site-package": "boto3" },
    {"site-package": "mypy_boto3_lambda"}
  ]
}
```

Results:
Screenshot 2021-11-23 at 6 04 44 PM

Fixes part of https://github.com/MLH-Fellowship/pyre-check/issues/58
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>

### CODE REVIEW

**Comment by pradeep90** [comment]
cc @0xedward @gbleaney @arthaud Do you have context here?

**Comment by abishekvashok** [comment]
@pradeep90 https://github.com/MLH-Fellowship/pyre-check/issues/58

https://github.com/facebook/pyre-check/pull/536

I think these should contain the things you are looking for, but if not sorry! Is there anything particular you are looking for?

File: stubs/third_party_taint/boto3_clients_sources_sinks.pysa [hunk 1]
```diff
@@ -22,3 +22,6 @@ def mypy_boto3_application_insights.client.ApplicationInsightsClient.list_log_pa
 def mypy_boto3_application_insights.client.ApplicationInsightsClient.list_log_patterns() -> TaintSource[ExceptionMessage]: ...
 def mypy_boto3_application_insights.client.ApplicationInsightsClient.list_problems() -> TaintSource[ExceptionMessage]: ...
 
+# AWS Lambda RemoteCodeExecution Sinks
+def mypy_boto3_lambda.client.LambdaClient.create_function(Code: TaintSink[RemoteCodeExecution], ImageConfig: TaintSink[RemoteCodeExecution]): ...
+def mypy_boto3_lambda.client.LambdaClient.update_function_code(ZipFile: TaintSink[RemoteCodeExecution], S3Bucket: TaintSink[RemoteCodeExecution], S3Key: TaintSink[RemoteCodeExecution], S3ObjectVersion: TaintSink[RemoteCodeExecution], ImageUri: TaintSink[RemoteCodeExecution]): ...
```
Comment (r0rshark) [inline stubs/third_party_taint/boto3_clients_sources_sinks.pysa:27]: @abishekvashok  why S3Key and S3ObjectVersion are sink for RCE? :)

**Review by r0rshark — CHANGES_REQUESTED** [review]

File: stubs/third_party_taint/boto3_clients_sources_sinks.pysa (hunk 1 above)
Comment (abishekvashok) [reply stubs/third_party_taint/boto3_clients_sources_sinks.pysa]: Ah... it's just me messing up 🤦  sorry. Reverted :)

**Comment by 0xedward** [comment]
Weird... It looks like this PR has been "merged" (https://github.com/facebook/pyre-check/commit/e6254a1c8a863f7615b4e50d83abd32ebb6ff91e) but the @facebook-github-bot didn't close this PR and comment about it.

Anyway, closing this PR because it has already been committed

**Pull request closed without merging**